src/SSHDebugPS/VS/VSOperationWaiter.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.SSHDebugPS.VS
+{
+    internal static class VSOperationWaiter
+    {
+        /// <summary>
+        /// Executes the specified action on a background thread, showing a wait dialog if it is available
+        /// </summary>
+        /// <param name="actionName">[Required] description of the action to show in the wait dialog</param>
+        /// <param name="action">[Required] action to run</param>
+        /// <param name="throwOnCancel">If specified, an OperationCanceledException is thrown if the operation is canceled</param>
+        /// <returns>True if the operation succeed and wasn't canceled</returns>
+        /// <exception cref="OperationCanceledException">Wait was canceled and 'throwOnCancel' is true</exception>
+        public static bool Wait(string actionName, bool throwOnCancel, Action action)
+        {
+            Task t = Task.Run(action);
+
+            WaiterImpl waiterImpl = null;
```

Comment:
>WaiterImpl [](start = 12, length = 10)

Now that we are in the MIEngine repo, we should add a TODO comment to make this host just use Microsoft.DebugEngineHost.HostWaitLoop